### PR TITLE
whitelist integration

### PR DESCRIPTION
**Pull request by davidangb**
download a whitelist from configurable GCS location; validate that current user is in the whitelist.

### CODE REVIEW

File: function/auth.js [hunk 1]
```diff
@@ -17,13 +18,18 @@ const requireAuthorizationHeader = (req) => {
...
-  // TODO: query whitelist for membership
+  // query whitelist for membership
+  const allowedUsers = await whitelist.readWhitelist(appConfig);
+  if (!allowedUsers.includes(email)) {
+    if (process.env.NODE_ENV !== 'test') {
+      console.error(`Rejecting request from unwhitelisted user ${email}.`);
+    }
+    // TODO: get verbiage/contact info from PO
+    throwHttpError(403,
+        'Access to this service is restricted. Contact [XXX] for more information.');
```
Comment (gpolumbo-broad) [inline function/auth.js:31]: Add contact info or add a `TODO`

File: function/config/default.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
   "samUrl": "https://sam.dsde-dev.broadinstitute.org/",
   "rawlsUrl": "https://rawls.dsde-dev.broadinstitute.org/",
-  "whitelistUrl": "gs://davidan-sts-test-whitelist-1/whitelist.json"
+  "whitelistBucket": "davidan-sts-test-whitelist-1",
```
Comment (gpolumbo-broad) [inline function/config/default.json:4]: Will testers/developers need to do anything in this bucket?  Are permissions set appropriately on this bucket to allow (or disallow) that interaction?

**Review by gpolumbo-broad — CHANGES_REQUESTED** [review]

File: function/auth.js (hunk 1 above)
Comment (davidangb) [reply function/auth.js:31]: the TODO is at line 29 :)

File: function/config/default.json (hunk 2 above)
Comment (davidangb) [reply function/config/default.json:4]: testers/developers don't need to do anything now, and the bucket is properly permissioned.  We are waiting on SIA approval and the final hosting projects to be created; once that's in place and deploy scripts are running,  I'll  go  back and update  all of this config.

**Review by gpolumbo-broad — APPROVED** [review]

**Comment by dvoet** [comment]
alternatively, you could use a group in sam for you whitelist

**Pull request merged**